### ISSUE TITLE
Using springdoc-openapi as BOM uses Gradle project verson rather than BOM version

### ISSUE THREAD

**Issue by mjustin** [springdoc/springdoc-openapi#1401]
## Describe the bug
I don't see it in the [project documentation](https://springdoc.org), but from #1099 & #1256, it looks like `springdoc-openapi` is designed to be used as a BOM, enforcing the dependency versions in the project to match the version of the BOM.

When I use this project as a BOM with Gradle using the [Spring dependency management plugin](https://docs.spring.io/dependency-management-plugin/docs/current/reference/html/), it uses the version of my Gradle project as the SpringDoc versions, rather than the version of the BOM.  For instance, if I have my Gradle project at version 0.0.1, use `org.springdoc:springdoc-openapi:1.6.1` as a BOM, and import `org.springdoc:springdoc-openapi-ui`, the version of `springdoc-openapi-ui` will be 0.0.1, not 1.6.1.

### `build.gradle`

```groovy
plugins {
    id 'org.springframework.boot' version '2.6.1'
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
}

group = 'com.example'
version = '0.0.1'
sourceCompatibility = '11'

repositories {
    mavenCentral()
}

dependencyManagement {
    imports {
        mavenBom 'org.springdoc:springdoc-openapi:1.6.1'
        mavenBom 'org.springframework.boot:spring-boot-dependencies:2.6.1'
    }
}

dependencies {
    implementation 'org.springframework.boot:spring-boot-starter'
    implementation 'org.springdoc:springdoc-openapi-ui'
}
```

### Execute

```
$ gradle compileJava

> Task :compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Could not resolve all files for configuration ':compileClasspath'.
   > Could not find org.springdoc:springdoc-openapi-ui:0.0.1.
     Required by:
         project :

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
1 actionable task: 1 executed
```

## Expected behavior
The expected behavior is that the version used for the `springdoc-openapi` BOM be used for the `org.springdoc` dependencies.

**Comment by bnasslahsen** [comment]
@mjustin, 

Review your syntax before logging bugs.

```gradle
dependencyManagement {
	imports {
		mavenBom('org.springdoc:springdoc-openapi:1.6.2') {
			bomProperty 'version', '1.6.2'
		}
	}
}
```

**Comment by mjustin** [comment]
@bnasslahsen 

Of all the BOM projects I've used in Gradle projects using Spring Boot (at least four, if not more), this is the only one I've ever had that didn't just work right out of the box without setting a `bomProperty`.  The syntax I've specified is exactly what I typically use, and is the syntax used in the first example in the [Dependency Management Plugin](https://docs.spring.io/dependency-management-plugin/docs/current/reference/html/#dependency-management-configuration-bom-import) documentation.  So I certainly wouldn't categorize this as a syntax error.

I do see in the documentation also covers [`bomProperty`](https://docs.spring.io/dependency-management-plugin/docs/current/reference/html/#dependency-management-configuration-bom-import-override).

That being said, it's highly surprising that `springdoc-openapi:1.6.2` requires setting the `version` properly in order to be used.  It looks like this mechanism is mostly intended to override a managed version (i.e. there's a standard sensible default that you want to use something different for).  I would expect that `springdoc-openapi:1.6.2` use `1.6.2` as the version by default.  Using the Gradle project's version would virtually never be appropriate, unless that Gradle project happened to be part of `springdoc-openapi`.

So with that all in mind, and knowing that the version property needs to be set in order to be used, I guess I'd want a bug or feature request that the version be defaulted to match the version of the BOM, rather than the Gradle project version of whatever project happens to be importing the BOM.
